### PR TITLE
samples: event_manager_proxy: Add nrf54h20pdk configuration

### PR DESCRIPTION
**Pull request by pdunaj**
Add configuration for nRF54H20PDK with PPR working as remote.

### CODE REVIEW

**Review by gmarull — CHANGES_REQUESTED** [review]

File: samples/event_manager_proxy/boards/nrf54h20pdk_nrf54h20_cpuapp_icmsg.conf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+CONFIG_UART_CONSOLE=y
+CONFIG_CONSOLE=y
+CONFIG_SERIAL=y
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply samples/event_manager_proxy/boards/nrf54h20pdk_nrf54h20_cpuapp_icmsg.conf]: I have dropped options as indeed there are no changes. For some reason I cannot remove the file as it causes error in cmake run.

File: samples/event_manager_proxy/boards/nrf54h20pdk_nrf54h20_cpuapp_icmsg.overlay [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+       status = "okay";
+};
+
+ipc0: &cpuapp_cpuppr_ipc {
+	status = "okay";
+};
+
+/* UART and RAM3 instance used by PPR should be enabled at build time
+ * using nordic-ppr snippet.
+ */
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply samples/event_manager_proxy/boards/nrf54h20pdk_nrf54h20_cpuapp_icmsg.overlay:25]: I want to keep it - even if it is documented somewhere I prefer to locally point to what is needed to save next person time.

File: samples/event_manager_proxy/remote/boards/nrf54h20pdk_nrf54h20_cpuppr_icmsg.conf [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+CONFIG_LOG=y
+CONFIG_LOG_MODE_MINIMAL=y
+CONFIG_ASSERT=n
+CONFIG_LOG_BACKEND_UART=n
+
+CONFIG_BOOT_BANNER=n
+CONFIG_PRINTK=y
+CONFIG_CONSOLE=y
+CONFIG_SERIAL=y
+CONFIG_UART_CONSOLE=y
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply samples/event_manager_proxy/remote/boards/nrf54h20pdk_nrf54h20_cpuppr_icmsg.conf]: done

File: samples/event_manager_proxy/remote/boards/nrf54h20pdk_nrf54h20_cpuppr_icmsg.overlay [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+ipc0: &cpuapp_cpuppr_ipc {
+	status = "okay";
+};
+
+mbox0: &cpuppr_vevif_local {
+	status = "okay";
+};
+
+mbox1: &cpuapp_bellboard {
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply samples/event_manager_proxy/remote/boards/nrf54h20pdk_nrf54h20_cpuppr_icmsg.overlay]: No really, only ipc0 is. I have added mbox as it looked symmetric to me this way.
Removed.

File: samples/event_manager_proxy/remote/boards/nrf54h20pdk_nrf54h20_cpuppr_icmsg.overlay [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+/* Enabled nodes required by IPC
+ * Two mboxes, one for each sides and one ipc instance
+ */
+
+ipc0: &cpuapp_cpuppr_ipc {
+	status = "okay";
+};
+
+mbox0: &cpuppr_vevif_local {
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply samples/event_manager_proxy/remote/boards/nrf54h20pdk_nrf54h20_cpuppr_icmsg.overlay]: done

**Comment by pdunaj** [comment]
rebased

File: samples/event_manager_proxy/sample.yaml [hunk 6]
```diff
@@ -1,17 +1,34 @@
...
     extra_args: CONF_FILE=prj_icmsg.conf
+  sample.event_manager_proxy.nrf54h20pdk_cpuapp.icmsg:
+    platform_allow:
+      - nrf54h20pdk_nrf54h20_cpuapp
+    integration_platforms:
+      - nrf54h20pdk_nrf54h20_cpuapp
+    extra_args:
+      event_manager_proxy_SNIPPET=nordic-ppr
+      event_manager_proxy_CONF_FILE=prj_icmsg.conf
+      event_manager_proxy_OVERLAY_CONFIG=boards/nrf54h20pdk_nrf54h20_cpuapp_icmsg.conf
```
Comment (gmarull) [inline samples/event_manager_proxy/sample.yaml:30]: can be dropped, together with the empty file

Comment (pdunaj) [reply samples/event_manager_proxy/sample.yaml]: missed that - thanks

**Review by gmarull — APPROVED** [review]

**Review by rakons — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

**Pull request merged**